### ISSUE TITLE
HTML node not extracting content

### ISSUE THREAD

**Issue by bakman2** [node-red/node-red#3137]
### Current Behavior

This appears version related. Example flow does not produce output under 2.0.6 when using HTML node.

### Expected Behavior

When running the flow, clicking the inject node, output should produce `Obligaties Wereldwijd`

### Steps To Reproduce

Run example flow on both version 1.3.7 and 2.0.6.

### Example flow

```
[{"id":"c5073e7167dfd3b0","type":"inject","z":"07c0e534b17ff9a0","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"<td data-th=\"Fonds\" class=\"name\">Obligaties Wereldwijd</td>","payloadType":"str","x":482,"y":144,"wires":[["6ee0d520cb8a2368","a91557144a38bd23"]]},{"id":"a91557144a38bd23","type":"debug","z":"07c0e534b17ff9a0","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":790,"y":144,"wires":[]},{"id":"6ee0d520cb8a2368","type":"html","z":"07c0e534b17ff9a0","name":"","property":"payload","outproperty":"payload","tag":".name","ret":"text","as":"single","x":628,"y":96,"wires":[["a91557144a38bd23"]]}]
```

### Environment

- Node-RED version: 1.3.7 and 2.0.6
- Node.js version:v14.17.6
- npm version: 6.14.15
- Platform/OS: LXC Debian container
- Browser: Safari 14.1.1 
-

**Comment by knolleary** [comment]
The html parsing library, `cheerio` had a major version upgrade as part of the 2.0 release.

That library made significant breaking changes in how it handles HTML fragments vs complete HTML documents. We tried to maintain backward compatibility - and got to the point our unit tests pass cleanly. However your scenario here does appear to have changed behaviour.

As you've noted in the Discourse thread, feeding in: `<td data-th="Fonds" class="name">Obligaties Wereldwijd</td>` does not work, but feeding in `<table><td data-th="Fonds" class="name">Obligaties Wereldwijd</td></table>` does.

I thought maybe the issue was trying to select on the top-level object, however passing in `<div class="name">Obligaties Wereldwijd</div>` does work as well. So I think this is specific to the `<td>` element. My guess is the library considers it an invalid fragment as the `<td>` tag cannot exist outside of a `<table>`.

Whilst it is not entirely satisfactory, I do not think there is anything we can do here. We needed to upgrade the cheerio library to pickup long-outstanding bug fixes - reverting it is not an option.
